official/resnet/resnet_run_loop.py

```diff
@@ -335,15 +385,8 @@ def resnet_main(flags, model_function, input_function, shape=None):
   # Using the Winograd non-fused algorithms provides a small performance boost.
   os.environ['TF_ENABLE_WINOGRAD_NONFUSED'] = '1'
 
-  if flags.multi_gpu:
-    validate_batch_size_for_multi_gpu(flags.batch_size)
-
-    # There are two steps required if using multi-GPU: (1) wrap the model_fn,
-    # and (2) wrap the optimizer. The first happens here, and (2) happens
-    # in the model_fn itself when the optimizer is defined.
-    model_function = tf.contrib.estimator.replicate_model_fn(
-        model_function,
-        loss_reduction=tf.losses.Reduction.MEAN)
+  # TODO(taylorrobie@): remove when per_device is no longer needed.
+  assign_multi_gpu(flags)
```

Comment:
How does this relate to per_device? 

Also, it appears flags.multi_gpu is only used in the line `   warn_on_multi_gpu_export(flags.multi_gpu)`. This can be changed to use flags.use_distribution_strategy, and the multi_gpu flag can be completely removed.

Reply:
It turns out we can probably remove multi_gpu altogether. Karmel just has to check that it doesn't break saved models.